### PR TITLE
Hide private methods - Closes #77

### LINKED ISSUE #77: API should hide private methods
There are several methods in the lisk-js API that aren't mentioned in the docs, look more like internal utility functions and shouldn't really be used by public, e.g. [sendRequestPromise](https://github.com/liskHQ/lisk-js/blob/development/lib/api/liskApi.js#L185
), getRandomPeer, banNode ...

If we expose these methods, people might use them in their projects and then we have to support them else we break other people's projects. On the other hand, if we make them private we can easily refactor and change their behaviour if necessary in the future.

There are several ways to emulate private methods in JavaScrtipt:
http://stackoverflow.com/questions/55611/javascript-private-methods

### PR DESCRIPTION
**Pull request by alepop**
Resolve #77 
- Expose private method to another module
- Rewrite tests

### CODE REVIEW

File: lib/api/privateApi.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+			'Content-Type': 'application/json',
+			'nethash': 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+			'broadhash': 'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
+			'os': 'lisk-js-api',
+			'version': '1.0.0',
+			'minVersion': '>=0.5.0',
+			'port': this.port
+		}
+	};
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: Please do not use semicolons after pure function declarations. We have not done this elsewhere in the codebase so far.

File: lib/api/privateApi.js [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+ * @private
+ */
+
+function getURLPrefix  () {
+	if (this.ssl) {
+		return 'https';
+	} else {
+		return 'http';
+	}
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: Semicolon, see above

File: lib/api/privateApi.js [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+function getFullUrl () {
+	var nodeUrl = this.currentPeer;
+
+	if (this.port) {
+		nodeUrl += ':'+this.port;
+	}
+
+	return getURLPrefix.call(this) + '://' + nodeUrl;
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+ * @private
+ */
+
+function getRandomPeer () {
+	var peers = (this.ssl) ? this.defaultSSLPeers : this.defaultPeers;
+	if (this.testnet) peers = this.defaultTestnetPeers;
+
+	var getRandomNumberForPeer = Math.floor((Math.random() * peers.length));
+	return peers[getRandomNumberForPeer];
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+		if (this.testnet) peers = this.defaultTestnetPeers;
+
+		for (var x = 0; x< peers.length; x++) {
+			if (this.bannedPeers.indexOf(currentRandomPeer) === -1) break;
+			currentRandomPeer = getRandomPeer.call(this);
+		}
+	}
+
+	return currentRandomPeer;
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+/**
+ * @method banNode
+ * @private
+ */
+
+function banNode () {
+	if (this.bannedPeers.indexOf(this.currentPeer) === -1) this.bannedPeers.push(this.currentPeer);
+	selectNode.call(this);
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+		} else {
+			reconnect = (peers.length !== this.bannedPeers.length);
+		}
+	// RandomPeer is not explicitly set, no peer discovery
+	} else {
+		reconnect = false;
+	}
+
+	return reconnect;
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+
+function checkOptions (options) {
+	Object.keys(options).forEach(function (optionKey) {
+		if (options[optionKey] === undefined || options[optionKey] !== options[optionKey]) {
+			throw { message: 'parameter value "'+optionKey+'" should not be '+ options[optionKey]  };
+		}
+	});
+
+	return options;
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+
+function serialiseHttpData (data) {
+	var serialised;
+
+	serialised = utils.trimObj(data);
+	serialised = utils.toQueryString(serialised);
+	serialised = encodeURI(serialised);
+
+	return '?'+serialised;
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+ * @param requestType
+ * @param options
+ * @private
+ *
+ * @return method string
+ */
+
+function checkRequest (requestType, options) {
+	return this.parseOfflineRequests(requestType, options).requestMethod;
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+			returnValue.requestMethod = transformRequest.requestMethod;
+			returnValue.requestParams = options;
+		}
+		break;
+	default:
+		break;
+	}
+
+	return returnValue;
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+ */
+
+function doPopsicleRequest (requestValue) {
+	return popsicle.request({
+		method: requestValue.requestMethod,
+		url: requestValue.requestUrl,
+		headers: requestValue.nethash,
+		body: requestValue.requestMethod !== 'GET' ? requestValue.requestParams : ''
+	}).use(popsicle.plugins.parse(['json', 'urlencoded']));
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

File: lib/api/privateApi.js [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+function sendRequestPromise (requestType, options) {
+	if (checkRequest.call(this, requestType, options) !== 'NOACTION') {
+		var requestValues = changeRequest.call(this, requestType, options);
+		return doPopsicleRequest.call(this, requestValues);
+	} else {
+		return new Promise(function (resolve) {
+			resolve({ done: 'done'});
+		});
+	}
+};
```
Comment (Tosch110) [inline lib/api/privateApi.js]: semicolon. s.o.

**Review by toschdev — CHANGES_REQUESTED** [review]
Overall, very good implementation of the privateAPI methods. I also like that you have created a new utils file for utility functions. Very good work, keep it up! Just few minor change requests for code format.

**Review by toschdev — DISMISSED** [review]

**Review by toschdev — DISMISSED** [review]

**Review by toschdev — APPROVED** [review]

**Pull request merged**